cmd/bootstrap/group_replication.go

```diff
@@ -239,20 +256,181 @@ func (m *mysqlsh) createCluster(ctx context.Context) error {
 	return nil
 }
 
-func (m *mysqlsh) addInstance(ctx context.Context, instanceDef string) error {
+type GTIDSetRelation string
+
+const (
+	GTIDSetEqual      GTIDSetRelation = "EQUAL"
+	GTIDSetContained  GTIDSetRelation = "CONTAINED"
+	GTIDSetContains   GTIDSetRelation = "CONTAINS"
+	GTIDSetDisjoint   GTIDSetRelation = "DISJOINT"
+	GTIDSetIntersects GTIDSetRelation = "INTERSECTS"
+)
+
+func gtidSubtract(ctx context.Context, shell *mysqlsh, a, b string) (string, error) {
+	a = strings.ReplaceAll(a, "\n", "")
+	b = strings.ReplaceAll(b, "\n", "")
+
+	query := fmt.Sprintf("SELECT GTID_SUBTRACT('%s', '%s') AS sub", a, b)
+
+	result, err := shell.runSQL(ctx, query)
+	if err != nil {
+		return "", errors.Wrapf(err, "execute %s", query)
+	}
+
+	return result.Rows[0]["sub"], nil
+}
+
+func compareGTIDs(ctx context.Context, shell *mysqlsh, a, b string) (GTIDSetRelation, error) {
+	if a == "" || b == "" {
+		if a == "" {
+			if b == "" {
+				return GTIDSetEqual, nil
+			}
+			return GTIDSetContained, nil
+		}
+
+		return GTIDSetContains, nil
+	}
+
+	aSubB, err := gtidSubtract(ctx, shell, a, b)
+	if err != nil {
+		return "", errors.Wrapf(err, "a sub b")
+	}
+
+	bSubA, err := gtidSubtract(ctx, shell, b, a)
+	if err != nil {
+		return "", errors.Wrapf(err, "b sub a")
+	}
+
+	if aSubB == "" && bSubA == "" {
+		return GTIDSetEqual, nil
+	} else if aSubB == "" && bSubA != "" {
+		return GTIDSetContained, nil
+	} else if aSubB != "" && bSubA == "" {
+		return GTIDSetContains, nil
+	} else {
+		query := fmt.Sprintf("GTID_SUBTRACT('%s', '%s')", a, b)
+		abIntersection, err := gtidSubtract(ctx, shell, a, query)
+		if err != nil {
+			return "", errors.Wrapf(err, "intersection")
+		}
+
+		if abIntersection == "" {
+			return GTIDSetDisjoint, nil
+		}
+
+		return GTIDSetIntersects, nil
+	}
```

Comment:
I think this logic can be simplified by splitting the else ifs to separate closes and leaving the else as a fallback.

I also think that `bSubA != ""` and `aSubB != ""` on the else if clauses can be omitted since the first clause guarantees that when it is false, either of them is different from an empty string.

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/1003/commits/59dae4e401dcc4418aa87eed9ff0746471115eee